sbin/winbindd-wrapper

```diff
@@ -0,0 +1,310 @@
+use Pod::Usage;
+use Fcntl qw(:flock);
+
+#pf::log must always be initilized first
+BEGIN {
+    # log4perl init
+    use constant INSTALL_DIR => '/usr/local/pf';
+    use lib INSTALL_DIR . "/lib";
+    use pf::log(service => 'winbindd-wrapper');
+}
+
+use pf::file_paths qw($var_dir);
+use pf::config qw(%Config %ConfigDomain);
+use pf::constants qw($FALSE $TRUE);
+use pf::services;
+use pf::util;
+use pf::services::util;
+use pf::StatsD;
+use pf::person;
+use Time::HiRes qw(time sleep);
+
+# initialization
+# --------------
+# assign process name (see #1464)
+our $PROGRAM_NAME = $0 = "winbindd-wrapper";
+our @REGISTERED_TASKS;
+our $IS_CHILD = 0;
+our %CHILDREN;
+our @TASKS_RUN;
+our $ALARM_RECV = 0;
+
+my $logger = get_logger( $PROGRAM_NAME );
+my $old_child_sigaction = POSIX::SigAction->new;
+
+POSIX::sigaction(
+    &POSIX::SIGCHLD,
+    POSIX::SigAction->new(
+        'child_sighandler' , POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    ),
+    $old_child_sigaction
+) or die("winbindd-wrapper could not set SIGCHLD handler: $!");
+
+POSIX::sigaction(
+    &POSIX::SIGALRM,
+    POSIX::SigAction->new(
+        'alarm_sighandler', POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or die("winbindd-wrapper could not set SIGALRM handler: $!");
+
+POSIX::sigaction(
+    &POSIX::SIGHUP,
+    POSIX::SigAction->new(
+        'reload_config' , POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or die("winbindd-wrapper could not set SIGHUP handler: $!");
+
+POSIX::sigaction(
+    &POSIX::SIGINT,
+    POSIX::SigAction->new(
+        'normal_sighandler' , POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or die("winbindd-wrapper could not set SIGINT handler: $!");
+
+POSIX::sigaction(
+    &POSIX::SIGTERM,
+    POSIX::SigAction->new(
+        'normal_sighandler' , POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or die("winbindd-wrapper could not set SIGTERM handler: $!");
+
+POSIX::sigaction(
+    &POSIX::SIGUSR1,
+    POSIX::SigAction->new(
+        'usr1_sighandler' , POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or die("winbindd-wrapper could not set SIGUSR1 handler: $!");
+
+my %args;
+getopts( 'dhvr', \%args );
+
+pod2usage( -verbose => 1 ) if ( $args{h} );
+
+my $daemonize = $args{d};
+my $verbose   = $args{v};
+my $restart   = $args{r};
+
+my $pidfile = "${var_dir}/run/winbindd-wrapper.pid";
+
+our $HAS_LOCK = 0;
+open(my $fh,">>$pidfile");
+flock($fh, LOCK_EX | LOCK_NB) or die "cannot lock $pidfile another winbindd-wrapper is running\n";
+$HAS_LOCK = 1;
+
+our $running = 1;
+our $process = 0;
+
+# standard signals and daemonize
+daemonize($PROGRAM_NAME) if ($daemonize);
+our $PARENT_PID = $$;
+
+
+sub start {
+    run_all_winbindd();
+}
+
+start();
+
+while($running){
+    sleep 1;
+    foreach my $pid (keys %CHILDREN){
+        unless(kill(0,$pid)){
+            $logger->error("Child $pid ($CHILDREN{$pid}) is dead. Respawning it.");
+            run_winbindd($CHILDREN{$pid});
+            delete $CHILDREN{$pid};
```

Comment:
Unfortunately I couldn't do that because the exec stays alive as a zombie and that made the detection not work given its still alive and I couldn't find a way to determine its a zombie.

If you have a way let me know, but that was the only way I could get it to work